compiler/damlc/tests/BUILD.bazel

```diff
@@ -199,54 +202,71 @@ da_haskell_library(
             ":integration-lib",
         ],
     )
-    for version, skip_validation in [(version, False) for version in COMPILER_LF_VERSIONS] + [
+    for version, skip_validation, eval_order in [(version, False, "LeftToRight") for version in COMPILER_LF_VERSIONS] + [
         (
             lf_version_configuration.get("default"),
             True,
+            "LeftToRight",
         ),
         (
             lf_version_configuration.get("dev"),
             True,
+            "LeftToRight",
+        ),
+        (
+            lf_version_configuration.get("dev"),
+            False,
+            "RightToLeft",
         ),
     ]
 ]
 
-da_haskell_test(
-    name = "integration-v1dev-script2",
-    size = "large",
-    srcs = ["src/DA/Test/DamlcIntegrationMain.hs"],
-    args = [
-        "--daml-lf-version",
-        "1.dev",
-        "--skip-validation",
-        "false",
-        "--daml-script-v2",
-        "true",
-    ],
-    data = [
-        ":bond-trading",
-        ":cant-skip-preprocessor",
-        ":daml-test-files",
-        ":package-vetting-package-a.dar",
-        ":package-vetting-package-b.dar",
-        ":query-lf-lib",
-        "//compiler/damlc/pkg-db",
-        "//compiler/damlc/stable-packages",
-        "//compiler/scenario-service/server:scenario_service_jar",
-        "//daml-script/daml3:daml3-script.dar",
-        "@jq_dev_env//:jq",
-        ghc_pkg,
-    ],
-    hackage_deps = [
-        "base",
-    ],
-    main_function = "DA.Test.DamlcIntegrationMain.main",
-    src_strip_prefix = "src",
-    visibility = ["//visibility:public"],
-    deps = [
-        ":integration-lib",
-    ],
-)
+[
+    da_haskell_test(
+        name = "integration{order}-v1dev-script2".format(
```

Comment:
I just realize that this is maybe a bit too costly to run all the tests twice with script v1 and v2.
Those integration tests take quite some time (almost 3min on my machine). 

Maybe we could test the evaluation order only on v2, or maybe on a selection of files from compiler/damlc/tests/daml-test-files.

Reply:
if we only test on v2, we should not allow this feature to be used with daml-script v1, even as early access IMO

Reply:
this feature will  be enable only for Daml 3.

Reply:
I have two questions:
- By v2 are you referring to v2 of the script runner here, or to something else?
- Do I currently understand that your suggestion is to keep [this for loop](https://github.com/digital-asset/daml/blob/444266edb2141b8addb319b902e8d63bd4c72d28/compiler/damlc/tests/BUILD.bazel#L265) but to get rid of `eval_order` in [this one](https://github.com/digital-asset/daml/blob/444266edb2141b8addb319b902e8d63bd4c72d28/compiler/damlc/tests/BUILD.bazel#L205)?

As long as we run all the integration tests once using each evaluation order that sounds good to me. What I want to avoid is ignoring some of them because as we've seen with the scala unit test, changing the evaluation order can have consequences in places you may not necessarily expect.

Reply:
> By v2 are you referring to v2 of the script runner here

Reply:
I meant reverting the change for the rule called "integration{skip}-v{version}" but keep the one for the rule called "integration-v1dev-script2".

The problem running the code twice, is that it costs you a "lot" for benefit which are debatable.
Let's take a step back a try to understand what we are actually testing.   
This involved: 
- The compiler,
- The script runner,
- The IDE leger.

1. First some tests involved only the compiler (because they check compilation error). We are already running those 2 times before your changes, now we are running those 4 times. 

2. For the tests that involves runner and the IDE legder, notice the only difference should be in the runner, the compiler and the IDE ledger should behave exactly in the same way in both test. So we are running a very long running tests, that involve a lot of moving peaces, that will be run any time we have a small change in the compiler, LF, or script to actually, to only verify the runner behave in the same way with both semantics. 

3. Finally your change will be enabled only for Daml 3.0, where we expect the runner v1 to be killed, so I do not see actual benefit to check that script v1 behave correctly with the new semantics. I

Reply:
As discussed on slack I think we're in agreement: let's limit the right-to-left tests to the -script2 target. I'll prepare a PR.

Reply:
https://github.com/digital-asset/daml/pull/17286 is the PR